crates/oci/src/client.rs

```diff
@@ -110,68 +128,29 @@ impl Client {
     /// if the digest cannot be determined).
     async fn push_locked_core(
         &mut self,
-        mut locked: LockedApp,
+        locked: LockedApp,
         auth: RegistryAuth,
         reference: Reference,
     ) -> Result<Option<String>> {
-        // For each component in the application, add a layer for the wasm module and
-        // separate layers for all static assets if application total will be under MAX_LAYER_COUNT,
-        // else an archive layer for all static assets per file entry if not.
-        // Finally, update the locked application with the layer digests.
-        let mut layers = Vec::new();
-        let mut components = Vec::new();
-        let archive_layers: bool = layer_count(locked.clone()).await? > MAX_LAYER_COUNT;
-
-        for mut c in locked.components {
-            // Add the wasm module for the component as layers.
-            let source = c
-                .clone()
-                .source
-                .content
-                .source
-                .context("component loaded from disk should contain a file source")?;
-
-            let source = parse_file_url(source.as_str())?;
-            let layer = Self::wasm_layer(&source).await?;
-
-            // Update the module source with the content ref of the layer.
-            c.source.content = Self::content_ref_for_layer(&layer);
-
-            layers.push(layer);
-
-            let mut files = Vec::new();
-            for f in c.files {
-                let source = f
-                    .content
-                    .source
-                    .context("file mount loaded from disk should contain a file source")?;
-                let source = parse_file_url(source.as_str())?;
-
-                if archive_layers {
-                    self.push_archive_layer(&source, &mut files, &mut layers)
-                        .await
-                        .context(format!(
-                            "cannot push archive layer for source {}",
-                            quoted_path(&source)
-                        ))?;
-                } else {
-                    self.push_file_layers(&source, &mut files, &mut layers)
-                        .await
-                        .context(format!(
-                            "cannot push file layers for source {}",
-                            quoted_path(&source)
-                        ))?;
-                }
-            }
-            c.files = files;
-            components.push(c);
+        let mut locked_app = locked.clone();
+        let mut layers = self
+            .assemble_layers(&mut locked_app, false)
```

Comment:
Perpetual Pet Peeve: a boolean here is not informative to the reader.  Consider a two-value enum e.g. `AssemblyMode::Simple` / `AssemblyMode::Consolidate` or `Archive::Yes` / `Archive::No` or whatever you feel would be most helpful to someone reading the code at the call site.

Reply:
Thanks for mentioning and advising; updated in https://github.com/fermyon/spin/pull/2395/commits/3871196285580fb1bbee0b14584738c644268036